urdf/CMakeLists.txt

```diff
@@ -29,13 +27,14 @@ install(FILES ${generated_compat_header} DESTINATION include/${PROJECT_NAME})
 add_library(${PROJECT_NAME} src/model.cpp)
 target_include_directories(${PROJECT_NAME}
   PUBLIC
+  ${urdfdom_INCLUDE_DIRS}
```

Comment:
One nitpick - this should use the [`urdfdom::urdfdom` target](https://github.com/ros2/urdfdom/blob/694a6fe34b4bc61652faea3fc02e06fc770cc802/urdf_parser/CMakeLists.txt#L70-L75) instead of the variables. This already looks to be taken care of by `ament_target_dependencies(urdf urdfdom ...)` below. What happens when this line using `urdfdom_INCLUDE_DIRS` is omitted?

Reply:
> What happens when this line using `urdfdom_INCLUDE_DIRS` is omitted?

It fails to build on macOS and Windows (though it succeeds just fine on Linux).

> One nitpick - this should use the [`urdfdom::urdfdom` target](https://github.com/ros2/urdfdom/blob/694a6fe34b4bc61652faea3fc02e06fc770cc802/urdf_parser/CMakeLists.txt#L70-L75) instead of the variables.

But that's a good point.  I'll give that a shot.

Reply:
> But that's a good point. I'll give that a shot.

Oh, right, now I remember.  This doesn't work because urdfdom is not actually an ament package.  Also, it uses the "old" style variables, not the new INTERFACE stuff.  Thus, we still have to do it with the `${urdfdom_INCLUDE_DIRS}` variable.  We should probably fix that, but I think that will come with a separate PR to urdfdom.

Reply:
> Oh, right, now I remember. This doesn't work because urdfdom is not actually an ament package. Also, it uses the "old" style variables, not the new INTERFACE stuff

Are you saying it because it doesn't use ament_cmake there's no `urdfdom_TARGETS` variable set, and so `ament_target_dependencies()` doesn't pick up on the target? [There still is a `urdfdom::urdfdom` target](https://github.com/ros2/urdfdom/blob/694a6fe34b4bc61652faea3fc02e06fc770cc802/urdf_parser/CMakeLists.txt#L70-L75). Modern style CMake would mean using `target_link_libraries(urdf PUBLIC urdfdom::urdfdom)` rather than the older `target_include_directories( urdfdom_INCLUDE_DIRS)`.

Reply:
> Are you saying because it doesn't use ament_cmake there's no `urdfdom_TARGETS` variable set, and so `ament_target_dependencies()` doesn't pick up on the target?

Yeah, exactly.

> [There still is a `urdfdom::urdfdom` target](https://github.com/ros2/urdfdom/blob/694a6fe34b4bc61652faea3fc02e06fc770cc802/urdf_parser/CMakeLists.txt#L70-L75). Modern style CMake would mean using `target_link_libraries(urdf PUBLIC urdfdom::urdfdom)` rather than the older `target_include_directories( urdfdom_INCLUDE_DIRS)`.

Hm, good point.  When I tried that, it didn't work.  Let me give it another shot.

Reply:
Ah, I see.  There is no `urdfdom::urdfdom` target, though there are `urdfdom::urdfdom_model`, `urdfdom::urdfdom_world`, etc.  I'm now doing `target_link_libraries` with them in 18883ea, let's see if Windows and macOS are happy.

Reply:
Ah I see. It seems like only `urdfdom::urdfdom_model` is needed: https://github.com/ros2/urdf/commit/1bbc0fe91e41261813be20b9f3a8c06473f1f9ba

It looks like CI is failing because `urdfdom` isn't exporting the transitive dependency on `urdfdom_headers`, which seems to be [because it uses `${urdfdom_headers_INCLUDE_DIRS}` instead of a target](https://github.com/ros2/urdfdom/blob/694a6fe34b4bc61652faea3fc02e06fc770cc802/CMakeLists.txt#L40). A workaround is to add `urdfdom_headers::urdfdom_headers` here: https://github.com/ros2/urdf/commit/ad2d430605ae1e80989d38540c0730d5487aaf23 . 

See https://ci.ros2.org/job/ci_linux/12309 vs https://ci.ros2.org/job/ci_linux/12310

Reply:
Upstream fix in https://github.com/ros2/urdfdom/pull/21